modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/persistence/RocksDbKeyValueStorage.java

```diff
@@ -703,7 +719,14 @@ public StatementResult invoke(If iif, HybridTimestamp opTs) {
                 if (branch.isTerminal()) {
                     Update update = branch.update();
 
-                    applyOperations(update.operations(), opTs);
+                    Collection<Operation> ops = new ArrayList<>(update.operations());
+
+                    ops.add(Operations.put(
+                            new ByteArray(ArrayUtils.concat(IDEMPOTENT_COMMAND_PREFIX_BYTES, ByteUtils.toBytes(commandId))),
+                            update.result().result())
```

Comment:
While restoring from snapshot, you assume that `update.result().result()` would always be 1-byte boolean. Lets add the assertion here.

Reply:
Please check my comment below.